docs/guide/advanced.md

```diff
@@ -0,0 +1,244 @@
+```bash
+$ npm run build
+$ npm version patch -m "build: bump version to %s"
+```
+
+### 5. Used during Umi UI
+
+Initialize the Umi application, install Umi UI and link the component library we just made:
+
+```bash
+$ mkdir umi-app && cd umi-app
+$ npx @umijs/create-dumi-app
+$ npm i @umijs/preset-ui -D
+$ npm link path/to/dumi/lib
+```
+
+In the `package.json` of the Umi application, manually add the component library as a dependency:
+
+```diff
+{
+  "dependencies": {
+    // Other dependencies
++   "your-lib-package-name": "*"
+  }
+}
+```
+
+Then start the Umi project as usual, you can see the demo assets in the dumi-lib project in the mini bubble of Umi UI, and can be directly inserted into the page for use:
+
+<p style="text-align: center;">
+  <img src="https://gw.alipayobjects.com/zos/bmw-prod/4102a494-e4d8-494e-a790-1a7a5562da51/kc6gnqjd_w680_h387.gif" width="680">
+</p>
+
+## Mobile component development
+
+The usage is very simple. After initializing the dumi project, install `dumi-theme-mobile` into `devDependencies`, and dumi will switch from the default PC component library development to mobile component development.
+
+<img style="border: 1px solid #eee;" src="https://gw.alipayobjects.com/zos/bmw-prod/acb29a94-6200-4798-82eb-190177fa841c/kezwf18r_w2556_h1396.jpeg" alt="Mobile theme preview effect" />
+
+Visit [Theme List - mobile](/theme#dumi-theme-mobile) to learn more about the features and HD solutions of mobile themes.
+
+## Automatic component API generation
+
+Now, we can realize the automatic generation of component API through JS Doc annotation + TypeScript type definition!
+
+### Types and annotations in component source code
+
+The premise of automatic component API generation is to ensure that dumi can deduce the content of the API through TypeScript type definition + annotations, such as the source code of the `Hello` component:
+
+```tsx | pure
+import React from 'react';
+
+export interface IHelloProps {
+  /**
+   * You can write the attribute description like this
+   * @description        You can also explicitly add a description name
+   * @description.zh-CN  Also supports different locale suffixes to achieve multi-language description
+   * @default            Support for defining default values
+   */
+  className?: string; // Support for identifying TypeScript optional types as optional attributes
+}
+
+const Hello: React.FC<IHelloProps> = () => <>Hello World!</>;
+
+export default Hello;
+```
+
+The type analysis tool behind dumi is `react-docgen-typescript`.
+
+For more types and annotation usage, please refer to [documentation](https://github.com/styleguidist/react-docgen-typescript#example).
+
+### Show the API in the documentation
+
+With the source code that can derive the API, we can render the API table through the built-in components of `API` in Markdown:
+
+```md
+<!-- Not passing src will automatically detect the current component, such as src/Hello/index.md will recognize src/Hello/index.tsx -->
```

Comment:
```suggestion
<!-- It will automatically detect the current component without src. For example, src/Hello/index.md will detect src/Hello/index.tsx -->
```